scaladoc/src/dotty/tools/scaladoc/tasty/TypesSupport.scala

```diff
@@ -290,10 +290,10 @@ trait TypesSupport:
         }
         inner(sc) ++ keyword(" match ").l ++ plain("{\n").l ++ casesTexts ++ plain(spaces + "}").l
 
-      case ParamRef(TypeLambda(names, _, _), i) => tpe(names.apply(i)).l
-
       case ParamRef(m: MethodType, i) => tpe(m.paramNames(i)).l ++ plain(".type").l
 
+      case ParamRef(binder: LambdaType, i) => tpe(binder.paramNames(i)).l
```

Comment:
`case ParamRef(TypeLambda(names, _, _), i)` was accidentally matching `TypeLambda` and `PolyType` here. Replaced with `LambdaType` that matches `TypeLambda`, `PolyType`, and `MethodType`.